arangod/RestHandler/RestSchemaHandler.cpp

```diff
@@ -0,0 +1,721 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2025-2025 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Business Source License 1.1 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Koichi Nakata
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Aql/Query.h"
+#include "Aql/QueryRegistry.h"
+#include "Basics/voc-errors.h"
+#include "Futures/Utilities.h"
+#include "Graph/GraphManager.h"
+#include "IResearch/IResearchLink.h"
+#include "IResearch/IResearchLinkHelper.h"
+#include "IResearch/IResearchView.h"
+#include "Logger/LogMacros.h"
+#include "RestHandler/RestSchemaHandler.h"
+#include "RestHandler/RestVocbaseBaseHandler.h"
+#include "Transaction/Methods.h"
+#include "Transaction/OperationOrigin.h"
+#include "Transaction/StandaloneContext.h"
+#include "Utils/CollectionNameResolver.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/Methods/Indexes.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/vpack.h>
+
+using namespace arangodb;
+using namespace arangodb::rest;
+
+namespace {
+constexpr std::string_view moduleName("graph management");
+}
+
+const std::string RestSchemaHandler::queryStr = R"(
+    LET samples = (
+      FOR d IN @@collection
+        SORT RAND()
+        LIMIT @sampleNum
+        RETURN UNSET(d, "_rev")
+    )
+    LET total = LENGTH(samples)
+
+    LET schemas = (
+      FOR d IN samples
+        LET keys = ATTRIBUTES(d)
+        FOR key IN keys
+          FILTER key != "_rev"
+          COLLECT attribute = key
+          AGGREGATE
+            count = COUNT(d),
+            types = UNIQUE(TYPENAME(d[key]))
+          RETURN {
+            attribute,
+            types,
+            optional: count < total
+          }
+    )
+    RETURN {
+      num: LENGTH(@@collection),
+      schemas: schemas,
+      examples: SLICE(samples, 0, @exampleNum)
+    }
+    )";
+
+RestSchemaHandler::RestSchemaHandler(ArangodServer& server,
+                                     GeneralRequest* request,
+                                     GeneralResponse* response,
+                                     aql::QueryRegistry* queryRegistry)
+    : RestCursorHandler(server, request, response, queryRegistry),
+      _graphManager(_vocbase, transaction::OperationOriginREST{::moduleName}) {}
+
```

Comment:
🤔 you are using the CollectionNameResolver now a couple of times.
I would build it here once same as the _graphManager.